### ISSUE TITLE
Files Null

### ISSUE THREAD

**Issue by michaelcuneo** [pqina/react-filepond#60]
### Summary

I have implemented the FilePond in my react project using the demo provided... the files upload, they show up on the preview window, but they don't update the state of the files in this.state.files... they are null every time.

### How to reproduce
My Constructor
```
  constructor(props) {
    super(props);
    
    // References created for each FilePond.
    this.featurePond = React.createRef();
    this.galleryPond = React.createRef();
    this.state = this.getInitialState();

    autoBind(this);
  }
```

Initialisation.
```
  handleInit() {
    console.log('FilePond feature instance has initialised', this.featurePond);
    console.log('FilePond gallery instance has initialised', this.galleryPond);
  }
```

The objects.
```
                <FilePond
                  ref={this.featurePond}
                  allowMultiple={false}
                  files={this.state.feature}
                  oninit={() => this.handleInit()}
                  onUpdateFiles={fileItems => {
                    this.setState({
                      feature: fileItems.map(fileItem => fileItem.file),
                    });
                  }}
                />
```

```
                <FilePond
                  ref={this.galleryPond}
                  allowMultiple
                  files={this.state.gallery}
                  maxFiles={10}
                  oninit={() => this.handleInit()}
                  onUpdateFiles={fileItems => {
                    this.setState({
                      gallery: fileItems.map(fileItem => fileItem.file),
                    });
                  }}
                />
```

If I console.log anywhere after a file has been added to the FilePond, it always prints undefined.

### Expected behaviour

Expecting it to actually store the state of my files that I want to upload, so that i can use the filename and such to do an GraphQL Upload to an Amazon S3.

### Additional information
    "react": "16.6.0",
    "filepond": "^4.1.0",
    "filepond-plugin-image-exif-orientation": "^1.0.4",
    "filepond-plugin-image-preview": "^4.0.1",

**Comment by michaelcuneo** [comment]
If I change

```
                  onUpdateFiles={fileItems => {
                    this.setState({
                      feature: fileItems.map(fileItem => fileItem.file),
                    });
                  }}
```

to

```
                  onUpdateFiles={fileItems => {
                    console.log(fileItems);
                  }}
```

When I drag a file onto the object, or browse, and it loads an image onto the preview window... there is no console output at all, it doesn't actually appear to be running onUpdateFiles at all.

**Comment by rikschennink** [comment]
Can you try `onupdatefiles` without the capitals.

Also, just so you know, you could do the uploading with FilePond as well.

**Comment by michaelcuneo** [comment]
Yeah, I solved my problem just now. I realised that because I type out all of my work to make sure it's all correct, and I learn more when I'm not copying and pasting... I automatically wrote the props as Camel Case. 

The uploading needs to happen alongside a GraphQL AppSync Complex object, so it's adding the Post, title, content, author, and feature image all in one go. The Feature image, being the S3 Upload.

I'd seperate this functionality, but I believe it's faster if I use the complex objects, soon as I figure out how, the documentation appears to be half done.

Thanks for the help.
